collections/pair.go

```diff
@@ -54,9 +54,22 @@ func PairKeyCodec[K1, K2 any](keyCodec1 codec.KeyCodec[K1], keyCodec2 codec.KeyC
 	}
 }
 
+// NamedPairKeyCodec instantiates a new KeyCodec instance that can encode the Pair, given the KeyCodec of the
+// first part of the key and the KeyCodec of the second part of the key, with names assigned to each part
+// which will only be used for indexing and informational purposes.
+func NamedPairKeyCodec[K1, K2 any](key1Name string, keyCodec1 codec.KeyCodec[K1], key2Name string, keyCodec2 codec.KeyCodec[K2]) codec.NamedKeyCodec[Pair[K1, K2]] {
```

Comment:
I was thinking that PairKeycodec can attempt to cast the codec to a named one? 

Would love to see how this might look in a module even in a throwaway branch

Reply:
See bank in #20547